assets/src/edit-story/components/panels/textStyle/font.js

```diff
@@ -47,16 +47,53 @@ function FontControls({ selectedElements, pushUpdate }) {
   const fontFamily = getCommonValue(selectedElements, 'fontFamily');
   const fontSize = getCommonValue(selectedElements, 'fontSize');
   const fontWeight = getCommonValue(selectedElements, 'fontWeight');
+  const timeout = useRef(null);
 
   const {
     state: { fonts },
-    actions: { getFontWeight, getFontFallback },
+    actions: { maybeEnqueueFontStyle, getFontWeight, getFontFallback },
   } = useFont();
   const fontWeights = useMemo(() => getFontWeight(fontFamily), [
     getFontWeight,
     fontFamily,
   ]);
 
+  const loadFont = useCallback(
+    (family, callback) => {
+      // 0. Check browser support to load fonts using JavaScript
+      if (document?.fonts) {
+        const font = `0 ${family}`;
+
+        // 1. Check if font family was not loaded yet
+        if (!document.fonts.check(font)) {
+          // 2. Try to inject <link /> resource of the font family
+          maybeEnqueueFontStyle(family);
+          // 3. Schedule a re-check since we need to wait the <link /> injection
+          timeout.current = setTimeout(async () => {
+            // 4. Load the new injected font in the client
+            await document.fonts.load(font);
+            // 5. Re-check if font was loaded
+            if (document.fonts.check(font)) {
+              return callback();
+            }
+            // 6. Try again if font still is not loaded
+            return loadFont(family, callback);
+          });
+        }
+      }
+
+      return callback();
+    },
+    [maybeEnqueueFontStyle]
```

Comment:
This should be moved elsewhere, I think. It seems like pretty generic functionality, that's relevant for anyone requesting a font to be loaded.